lib/config.py

```diff
@@ -77,62 +79,88 @@ class ConfigParser(object):
     Precedence is:
     cmdline > config file > argparse defaults
     """
-    def __init__(self):
-        self._CONF = self._parse_arguments()
+    def __init__(self, current_working_dir=None):
+        self.parser = argparse.ArgumentParser()
+        self._CONF = None
+        if current_working_dir:
+            self._cwd = current_working_dir
+        else:
+            self._cwd = os.getcwd()
+
+        self._setup_config_parser_options()
 
     @property
     def CONF(self):
         return self._CONF
 
-    def _parse_config(self, config_file):
+    def _setup_config_parser_options(self):
+        """
+	Configures the argument parser object to match the expected
+            configuration.
+        """
+        self.parser.add_argument('--config-file', '-c',
+                                 help='Path of the configuration file for build '
+                                     'scripts',
+                                 # NOTE(maurosr): move this to /etc in the future
+                                 default='./config.yaml')
+        self.parser.add_argument('--packages', '-p',
+                                 help='Packages to be built',
+                                 nargs='*')
+        self.parser.add_argument('--log-file', '-l',
+                                 help='Log file',
+                                 default='/var/log/host-os/builds.log')
+        self.parser.add_argument('--verbose', '-v',
+                                 help='Set the scripts to be verbose',
+                                 action='store_true')
+        self.parser.add_argument('--result-dir', '-r',
+                                 help='Directory to save the RPMs.',
+                                 default=os.path.join(self._cwd, 'result'))
```

Comment:
> TBH, I don't like the idea of our code implicitly converting paths to absolute paths in these cases.

What's the problem? The resulting path will be pointing to the same place. Anyway, I've checked and using a relative path for result_dir does not cause problems, so we could just leave the relative path.
The problem I see with the current implementation is the different behavior from command line and config file.
If current_working_dir="/foo" is passed to the constructor and '--result-dir=bar' is passed to the command line, result_dir will be "/foo/bar in the resulting dict.
However, if result_dir is instead passed as 'bar' in the config file, the constructor argument will not be prepended, and result_dir will be just "bar in the resulting dict.
I think this "os.path.join(self._cwd," should just be removed.